### ISSUE TITLE
[BUG] access discriminant in return aggregate would be a dangling reference

### ISSUE THREAD

**Issue by pyjarrett** [pyjarrett/dir_iterators#2]
Getting this after switching to using this with FSF GNAT.

```ada
    function Element_Value
       (Tree : Recursive_Dir_Walk; Position : Cursor) return Reference_Type is
    begin
        pragma Unreferenced (Tree);
        return Reference_Type'(Element => Position.It.Next_Entry'Access);  -- HERE
    end Element_Value;
```

**Comment by simonjwright** [comment]
Compiles OK with FSF GCC 11.1.0 (x86_64-apple-darwin)

**Comment by pyjarrett** [comment]
Fixed in efeb352dab2196d0dedb44ad82b33c17600c43bd.
